### PR TITLE
Fixed #33670 -- Fixed altering primary key on SQLite.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-33670

### CODE REVIEW

File: django/db/backends/sqlite3/schema.py [hunk 1]
```diff
@@ -216,8 +216,9 @@ def is_self_referential(f):
         if getattr(create_field, "primary_key", False) or (
             alter_field and getattr(alter_field[1], "primary_key", False)
         ):
+            _, new_field = alter_field
             for name, field in list(body.items()):
-                if field.primary_key:
+                if field.primary_key and name != new_field.name:
```
Comment (felixxm) [inline django/db/backends/sqlite3/schema.py:221]: Do not remove the old primary key when an altered field that introduces a primary key is the same field.

**Comment by felixxm** [comment]
This can also fix ticket-29790 and ticket-28541 :thinking:

File: tests/schema/tests.py [hunk 2]
```diff
@@ -3584,6 +3584,21 @@ def test_primary_key(self):
...
+
+        old_field = Thing._meta.get_field("when")
+        new_field = CharField(max_length=2, primary_key=True)
+        new_field.set_attributes_from_name("when")
+        new_field.model = Thing
+        with connection.schema_editor() as editor:
+            editor.alter_field(Thing, old_field, new_field, strict=True)
+        self.assertEqual(self.get_primary_key(Thing._meta.db_table), "when")
+        with connection.schema_editor() as editor:
+            editor.alter_field(Thing, new_field, old_field, strict=True)
```
Comment (felixxm) [inline tests/schema/tests.py:3599]: Without a fix this tries to create a table with implicit primary key:
```
CREATE TABLE "new__drop" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "when" varchar(1) NOT NULL)
```

**Review by David-Wobrock — APPROVED** [review]
Looks good to me, thank you :+1:

**Review by carltongibson — APPROVED** [review]
Yep. Good spot. Thanks @felixxm

File: django/db/backends/sqlite3/schema.py (hunk 1 above)
Comment (timgraham) [reply django/db/backends/sqlite3/schema.py:221]: Maybe that's worth a code comment.

Comment (felixxm) [reply django/db/backends/sqlite3/schema.py:221]: I added a comment.

**Comment by felixxm** [comment]
Thanks y'all for reviews :star2:

**Pull request merged**